File: TRAINSET.py
t=int(input())

while(t!=0):
	n=int(input())
	ls=[]
	for i in range(n):
		ls.append(list(input()))
		
	arr=[]
	for i in range(n):
		arr.append(int (ls[i][-1]))
	# ~ print(arr)
	ones=arr.count(1)
	zeroes=arr.count(0)
	ans=max(ones,zeroes)
	if(ans==ones):
		print(ans)
	else:
		
		if(ans%2==0):
			print(ans)
		else:
			print(ans+1)
	t-=1

File: TRAINSET.c
#include<stdio.h>
int max(int a,int b)
{
	if(a>b)
	{
		return a;
	}
	else
	{
		return b;
	}
}

void solve()
{
	int n;
	scanf("%d",&n);
	int arr[25000];
	int s[1000];
	for(int i=0;i<n;i++)
	{
		int l=scanf("%[^\n]%*c",s);
		arr[i]=(int)(s[l-1]);
	}
	for(int i=0;i<n;i++)
	{
		printf("%d ",arr[i]);
	}
	int zeroes=0;
	int ones=0;
	for(int i=0;i<n;i++)
	{
		if(arr[i]==0)
		{
			zeroes++;
		}
		else
		{
			ones++;
		}
	}
	if(max(ones,zeroes)%2==0)
	{
		printf("%d",max(ones,zeroes));
	}
	else
	{
		printf("%d",max(ones,zeroes)+1);
	}
}
int main()
{
	int t=0;
	scanf("%d",&t);
	while(t--)
	{
		solve();
	}
	return 0;
}
